### PR TITLE
For #430: TableListView test with authentication.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #430: `TableListView` test with authentication.
- Fixed test for `TableListView` mocking authenticated user related to `Company`, `Floor` and `Location` (to make sure that the user has permission to some group of tables)
- Left puzzle for fixing `TableListView`, which is broken: it just returns the header of the page

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#506](https://github.com/timelesslounge/timelessis/issues/506) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#506](https://github.com/timelesslounge/timelessis/issues/506) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/ece1317a-a0be-4eab-8931-f3c8141b53bc); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/view/tables/test_tables.py [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+    company = factories.CompanyFactory()
+    employee = factories.EmployeeFactory(
+     company=company, role_id=role.id
+    )
+    location = factories.LocationFactory(company=company)
+    floor = factories.FloorFactory(location=location)
+    with client.session_transaction() as session:
+        session["user_id"] = employee.id
+    g.user = employee
+    factories.TableFactory(floor_id=floor.id, name="Table 01")
```
Comment (vryazanov) [inline tests/view/tables/test_tables.py]: @paulodamaso Factory provides method for bulk creation:
`factories.TableFactory.create_batch(size=3, floor_id=floor.id, name="Table 01")`

File: tests/view/tables/test_tables.py [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+    factories.TableFactory(floor_id=floor.id, name="Table 01")
+    factories.TableFactory(floor_id=floor.id, name="Table 01")
     response = client.get("/tables/")
+    print(response.data)
```
Comment (vryazanov) [inline tests/view/tables/test_tables.py]: @paulodamaso let's remove print :)

File: tests/view/tables/test_tables.py [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
+    )
+    location = factories.LocationFactory(company=company)
+    floor = factories.FloorFactory(location=location)
+    with client.session_transaction() as session:
+        session["user_id"] = employee.id
+    g.user = employee
+    factories.TableFactory(floor_id=floor.id, name="Table 01")
+    factories.TableFactory(floor_id=floor.id, name="Table 01")
+    factories.TableFactory(floor_id=floor.id, name="Table 01")
     response = client.get("/tables/")
```
Comment (vryazanov) [inline tests/view/tables/test_tables.py]: @paulodamaso let's add `url_for` instead of url. wdyt?

**Comment by paulodamaso** [comment]
@vladarefiev Thanks for the comments, please take a look at the corrections. About the `factories.TableFactory.create_batch(size=3, floor_id=floor.id, name="Table 01")` usage, I can't use it because I have to know the table names to assert if they are correct later

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#506](https://github.com/timelesslounge/timelessis/issues/506) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#506](https://github.com/timelesslounge/timelessis/issues/506) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@vryazanov ping
@emilianodellacasa I believe you will need to say "quality acceptable" here. Thanks

**Comment by vryazanov** [comment]
@ypshenychka @emilianodellacasa I confirm

**Comment by emilianodellacasa** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by ypshenychka** [comment]
@emilianodellacasa It's still in my agenda

**Comment by emilianodellacasa** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/timelesslounge/timelessis/issues/506#issuecomment-473508127))

@emilianodellacasa Job `gh:timelesslounge/timelessis#506` is not assigned, can't get performer

**Comment by emilianodellacasa** [comment]
@ypshenychka now?

**Comment by ypshenychka** [comment]
@emilianodellacasa Still on the list...

**Comment by emilianodellacasa** [comment]
@0crat out

**Comment by emilianodellacasa** [comment]
@ypshenychka going trial-and-error here :-) now?

**Comment by 0crat** [comment]
> @0crat out  ([here](https://github.com/timelesslounge/timelessis/issues/506#issuecomment-473509817))

@emilianodellacasa Job `gh:timelesslounge/timelessis#506` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope

**Comment by emilianodellacasa** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/506#issuecomment-473510692))

@emilianodellacasa This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#506](https://github.com/timelesslounge/timelessis/issues/506) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23506%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by emilianodellacasa** [comment]
@ypshenychka should be ok now, confirm?

**Comment by ypshenychka** [comment]
@emilianodellacasa Still there
image

**Comment by emilianodellacasa** [comment]
@ypshenychka can you wait the issue?

**Comment by ypshenychka** [comment]
@emilianodellacasa It's not bothering me, but would be nice if it goes away from my agenda someday :)

**Comment by emilianodellacasa** [comment]
@ypshenychka It would be better to open a new issue on zerocracy and ask for advice to solve current situation, what do you think?

**Comment by ypshenychka** [comment]
@emilianodellacasa Good idea. Will do. Thanks

**Comment by ypshenychka** [comment]
@emilianodellacasa BTW my agenda is now clear. Not sure how it happened, but issue is not actual anymore :)

**Pull request merged**